smack-core/src/main/java/org/jivesoftware/smack/remoteEndpoint/HttpLookUp.java

```diff
@@ -0,0 +1,197 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jivesoftware.smack.remoteEndpoint;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.HttpURLConnection;
+import java.net.InetAddress;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+import java.nio.charset.Charset;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+
+import org.jivesoftware.smack.datatypes.UInt16;
+import org.jivesoftware.smack.util.PacketParserUtils;
+import org.jivesoftware.smack.util.rce.RemoteConnectionEndpoint;
+import org.jivesoftware.smack.xml.XmlPullParser;
+import org.jivesoftware.smack.xml.XmlPullParserException;
+
+import org.minidns.util.InetAddressUtil;
+
+/**
+ * The HTTP lookup method uses web host metadata to list the URIs of alternative connection methods.<br>
+ * <br>
+ * To obtain endpoints for Bosh or Websocket connections, use {@link RelAttribute#BOSH} and <br>
+ * {@link RelAttribute#WEBSOCKET} respectively with {@link #getEndpoints(RelAttribute)}. <br>
+ * <br>
+ * Example:<br>
+ * <pre>
+ * {@code
+ * HttpLookUp lookup = new HttpLookUp(host,usingHttps);
+ * List<RemoteConnectionEndpoint> endpoints = lookup.getEndpoints(RelAttribute.WEBSOCKET);
+ * }
+ * </pre>
+ * @see <a href="https://xmpp.org/extensions/xep-0156.html#http">
+*     HTTP Lookup Method.
+*     </a>
+ */
+public final class HttpLookUp {
+
+    private static final int BUFFER_SIZE = 256;
+    private static final String XRD_NAMESPACE = "http://docs.oasis-open.org/ns/xri/xrd-1.0";
+
+    /**
+     * Specifies a link relation for the selected type of connection.
+     */
+    enum RelAttribute {
+        /**
+         * Selects link relation attribute as "urn:xmpp:alt-connections:xbosh".
+         */
+        BOSH("urn:xmpp:alt-connections:xbosh"),
+        /**
+         * Selects link relation attribute as "urn:xmpp:alt-connections:websocket".
+         */
+        WEBSOCKET("urn:xmpp:alt-connections:websocket");
+        private final String attribute;
+        RelAttribute(String relAttribute) {
+            this.attribute = relAttribute;
+        }
+    }
+
+    private final int content_length;
+    private final HttpURLConnection httpConnection;
+
+    String obtainedXml = "";
+
+    /**
+     * Constructs a HTTP connection with the server and retrieves complete host metadata.
+     *
+     * @param host host that needs to be connected
+     * @param usingHttps true if secure HTTP being used, false otherwise
+     * @throws IOException exception due to input/output operations
+     */
+    public HttpLookUp(String host, boolean usingHttps) throws IOException {
```

Comment:
Don't throw exceptions in the constructor.

Reply:
An idea: You might also think about removing the usingHttps argument and instead let your implementation optimistically try https first and then fall back to http?

Reply:
An idea: You might also think about removing the usingHttps argument and instead let your implementation optimistically try https first and then fall back to http?

Reply:
As discussed, should only perform https lookups.